server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -522,19 +522,10 @@ private boolean shouldReverseLeafReaderContexts() {
         // This is actually beneficial for search queries to start search on latest segments first for time series workload.
         // That can slow down ASC order queries on timestamp workload. So to avoid that slowdown, we will reverse leaf
         // reader order here.
-        if (searchContext.indexShard().isTimeSeriesDescSortOptimizationEnabled()) {
-            // Only reverse order for asc order sort queries
-            if (searchContext.sort() != null
-                && searchContext.sort().sort != null
-                && searchContext.sort().sort.getSort() != null
-                && searchContext.sort().sort.getSort().length > 0
-                && searchContext.sort().sort.getSort()[0].getReverse() == false
-                && searchContext.sort().sort.getSort()[0].getField() != null
-                && searchContext.sort().sort.getSort()[0].getField().equals(DataStream.TIMESERIES_FIELDNAME)) {
-                return true;
-            }
-        }
-        return false;
+        return searchContext.indexShard().isTimeSeriesDescSortOptimizationEnabled()
```

Comment:
```suggestion
        return searchContext.shouldUseTimeSeriesDescSortOptimization();
```

Reply:
Oops I missed removing this method entirely